src/Npgsql/TypeMapping/TypeMapper.cs

```diff
@@ -103,31 +102,33 @@ internal bool TryResolveByOID(uint oid, [NotNullWhen(true)] out NpgsqlTypeHandle
         if (_handlersByOID.TryGetValue(oid, out handler))
             return true;
 
-        if (!DatabaseInfo.ByOID.TryGetValue(oid, out var pgType))
-            return false;
+        return TryResolveLong(oid, out handler);
 
-        lock (_writeLock)
+        bool TryResolveLong(uint oid, [NotNullWhen(true)] out NpgsqlTypeHandler? handler)
         {
-            if ((handler = ResolveByDataTypeNameCore(pgType.FullName)) is not null)
+            if (!DatabaseInfo.ByOID.TryGetValue(oid, out var pgType))
             {
-                _handlersByOID[oid] = handler;
-                return true;
+                handler = null;
+                return false;
             }
-            
-            if ((handler = ResolveByDataTypeNameCore(pgType.Name)) is not null)
-            {
-                _handlersByOID[oid] = handler;
-                return true;
-            }
-            
-            if ((handler = ResolveComplexTypeByDataTypeName(pgType.FullName, throwOnError: false)) is not null)
+
+            lock (_writeLock)
             {
-                _handlersByOID[oid] = handler;
-                return true;
-            }
+                if ((handler = ResolveByPostgresType(pgType)) is not null)
```

Comment:
Yeah, that's intentional. The reason we were doing it twice it because some resolvers (like builtin) don't care about FullName. I'm pretty sure none of out resolvers actually need FullName.
The point with `ResolveByPostgresType` is that we pass the type to the resolver, which will decide whether it can handle this type or not. To make it a bit easier, by default resolver's `ResolveByPostgresType` just calls `ResolveByDataTypeName` with a short name, and users have an option to override this.
As for the caching, we cache by FullName because it's unique (technically more correct is to cache by `PgType`, but this looks like an overkill).

Reply:
> I'm pretty sure none of out resolvers actually need FullName.

Yeah, that makes sense. The only types which need a full name (i.e. in non-pg_catalog schemas) are extensions, and those can be in any schema, based on how the user adds the extension.